src/lib/mina_base/test/zkapp_command/fee_related.ml

```diff
@@ -0,0 +1,37 @@
+open Core_kernel
+open Mina_base.Zkapp_command
+open Mina_base
+open Snarky_backendless
+open Snark_params.Tick
+
+let%test_module "fee_related_tests" =
+  ( module struct
+    let feepayer_body_generator = Zkapp_command.T.Stable.Latest.Wire.gen
+
+    let%test_unit "fee" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: Currency.Fee.t] x.fee_payer.body.fee
+            (fee @@ Zkapp_command.T.Stable.Latest.of_wire x) )
+
+    let%test_unit "fee_payer_account_update" =
+      Quickcheck.test ~trials:50 feepayer_body_generator ~f:(fun x ->
+          [%test_eq: Account_update.Fee_payer.Stable.V1.t] x.fee_payer
```

Comment:
`Account_update.Fee_payer.t`

Avoid using versioned modules, they're only for defining types